staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -272,8 +274,15 @@ func (s *Controller) processNextWorkItem() bool {
 		return true
 	}
 
-	runtime.HandleError(fmt.Errorf("error processing service %v (will retry): %v", key, err))
-	s.queue.AddRateLimited(key)
+	var re *api.RetryError
+	if stderrors.As(err, &re) {
```

Comment:
What method of LoadBalancer are we expecting to return RetryError? Asking because many of the errors are wrapped and not sure this check will work even when RetryError is returned.

Reply:
> For now, the PR is meant to provide a starting point for discussing. Hence, I have not invested into adding tests. Once/if we have consent on the general direction of this PR, I am going to complete the missing pieces.

Ah sorry, I missed this from the PR description. This seems pretty reasonable to me. @cheftako @bowei @MrHohn thoughts?

Reply:
>What method of LoadBalancer are we expecting to return `RetryError`?

To add to that: for now, `RetryError` responses are only honored from `EnsureLoadBalancer` calls. It would be trivial to also do so for `GetLoadBalancer` and `EnsureLoadBalancerDeleted` (by replacing `%v` with `%w` in the right places as well), though I am not sure how useful that might be. It definitely isn't for DigitalOcean, so I tried to keep things minimalistic for now.

Reply:
+1, enabling cloud providers to have control over how the retry should happen sounds reasonable to me as well.